packages/block-editor/src/store/reducer.js

```diff
@@ -1228,6 +1247,7 @@ export default combineReducers( {
 	isTyping,
 	isCaretWithinFormattedText,
 	blockSelection,
+	blockZoom,
```

Comment:
If I create a group block, add some blocks inside it, select all the blocks inside, zoom them, and then press the undo button the editor crashes.
I guess blockZoom probably needs to be inside blocks reducer branch, so it is covered by undo functionality. As a side note, we should decide if zooming in creates a undo level or not, there are pros and cons of either option. But it is possible to have the functionality working correctly with undos without creating undo levels, like the block selection done in https://github.com/WordPress/gutenberg/pull/16428.

Related https://github.com/WordPress/gutenberg/issues/12327.

Reply:
Yes I need to move it, but I don't think it should create an undo level. It's not an edit operation, it's more like selections which we also don't expect to create undo levels.

I plan to give this reducer it's own history so that you can zoom out in steps after zooming in multiple times.

I wonder if it'll be easy to nest 2 reducers with history like that.